src/batMap.mli

```diff
@@ -86,6 +86,12 @@ sig
       [m], plus a binding of [x] to [y]. If [x] was already bound
       in [m], its previous binding disappears. *)
 
+  val update: key -> key -> 'a -> 'a t -> 'a t
+  (** [update k1 k2 v2 m] replace the previous binding of [k1] in [m] by
+      [k2] associated to [v2].
```

Comment:
I would add the following sentence: "This is equivalent to [add k2 v2 (remove k1) m], but more efficient in the case where `k1` and `k2` have the same key ordering."